### PR TITLE
server-acl-init: Create global ACL auth method for API Gateway in secondary dc

### PR DESCRIPTION
**Pull request by nathancoleman**
This is a followup for #1462 

**Changes proposed in this PR:**
API Gateway needs to discover services in other datacenters (see #1344 and #1479), so
- Use global ACL policy for `api-gateway-controller` instead of local
- Use correct auth method name for binding when deployed in secondary datacenter

**How I've tested this PR:**
Install primary and secondary Consul clusters as described in [Federation Between Kubernetes Clusters](https://www.consul.io/docs/k8s/deployment-configurations/multi-cluster/kubernetes), making sure to add to the Consul configuration to enable API Gateway and use the control plane image from this PR:
```yaml
global:
  imageK8S: "hashicorpdev/consul-k8s-control-plane:f1a9304a"
...

apiGateway:
  enabled: true
  image: "hashicorp/consul-api-gateway:0.4.0"
```

The difference that we're looking for is that the API Gateway controller will not complete its init process w/ the current `main` branch but will with these changes pinned for `global.imageK8S`. This is because the global auth method exists with the right policies and bindings now due to the changes in the init job.

> **Note**
> The API Gateway controller still will not be able to create functional Gateways in the secondary datacenter. This is due to the issue described in https://github.com/hashicorp/consul-api-gateway/issues/361; however, the changes in this PR fix the acl-init which is half of the problem.

**How I expect reviewers to test this PR:**
See above

**Checklist:**
- [x] Tests added
- [x] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

**Comment by manobi** [comment]
I've tried with `hashicorpdev/consul-k8s-control-plane:f1a9304a`. 
It fixes the `api-gateway-controller-acl-init` issue, but `api-gateway-controller` keeps restarting with the following log message:

```
2022-09-07T14:09:34.911Z [ERROR] consul/certmanager.go:253: consul-api-gateway-server.cert-manager: error grabbing leaf certificate: error="Unexpected response code: 403 (rpc error making call: rpc error making call: Permission denied: token with AccessorID 'REDACTED' lacks permission 'service:write' on \"consul-api-gateway-controller\")"
```

**Comment by nathancoleman** [comment]
@manobi that matches what I was seeing while testing yesterday. Have been discussing internally what all it will take to get the setup with a controller per cluster working, and this is the first of 2-3 I believe. The error that the controller is logging I believe will require a followup PR over in the consul-api-gateway repo. I will pursue merge of this change and then spend some time today working on that followup PR.

**Comment by manobi** [comment]
@nathancoleman Thank you for all the effort so far.
I'm happy to confirm that this PR fixes the original issue [1479](https://github.com/hashicorp/consul-k8s/issues/1479).

If you need help to test something in consul-api-gateway repo just ping me.

**Comment by nathancoleman** [comment]
I've identified the next couple of issues preventing Consul API Gateway from working in federated setups with a controller per datacenter. Still working to determine scope and priority.

In the meantime, we've added a section to our docs describing the current limitations of API Gateway in federated setups [here](https://www.consul.io/docs/api-gateway/tech-specs#limited-support-of-some-consul-features)

**Review by andrewstucki — APPROVED** [review]
This looks fine as far as I can tell, but I'd want someone who is a bit more familiar with the acl-init code to sign off before merging.

File: control-plane/subcommand/server-acl-init/command.go [hunk 1]
```diff
@@ -592,7 +592,15 @@ func (c *Command) Run(args []string) int {
 			return 1
 		}
 		serviceAccountName := c.withPrefix("api-gateway-controller")
-		if err := c.createACLPolicyRoleAndBindingRule("api-gateway-controller", rules, consulDC, primaryDC, localPolicy, primary, localComponentAuthMethodName, serviceAccountName, consulClient); err != nil {
+
+		// API gateways require a global policy/token because they must
+		// discover services in other datacenters.
```
Comment (mikemorris) [inline control-plane/subcommand/server-acl-init/command.go:597]:
```suggestion
		// create config-entry resources in the primary, even when deployed to a secondary datacenter
```
Isn't ^ the actual reason why this is necessary? We aren't enabling routing to services in other datacenters yet.

Comment (nathancoleman) [reply control-plane/subcommand/server-acl-init/command.go:597]: You're right, good catch!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 ## UNRELEASED
 
+BUG FIXES:
+* Control plane
+  * Create global ACL auth method for API Gateway in secondary datacenter [[GH-1481](https://github.com/hashicorp/consul-k8s/pull/1481)]
```
Comment (kschoche) [inline CHANGELOG.md:5]:
```suggestion
  * Use global ACL auth method to provision ACL tokens for API Gateway in secondary datacenter [[GH-1481](https://github.com/hashicorp/consul-k8s/pull/1481)]
```

**Review by kschoche — APPROVED** [review]
Looks great!!!
I left one suggestion in the changelog to be a bit more accurate in the description of the issue.

**Pull request merged**